llama.h

```diff
@@ -0,0 +1,64 @@
+#pragma once
+
+#include <string>
+#include <vector>
+#include <map>
+#include <memory>
+
+#include "utils.h"
+#include "ggml.h"
+
+#ifdef LLAMA_SHARED
+#    ifdef _WIN32
+#        ifdef LLAMA_BUILD
+#            define LLAMA_API __declspec(dllexport)
+#        else
+#            define LLAMA_API __declspec(dllimport)
+#        endif
+#    else
+#        define LLAMA_API __attribute__ ((visibility ("default")))
+#    endif
+#else
+#    define LLAMA_API
+#endif
+
+
+
+
+// default hparams (LLaMA 7B)
+struct llama_hparams {
+    int32_t n_vocab = 32000;
+    int32_t n_ctx   = 512;   // this is provided as user input?
+    int32_t n_embd  = 4096;
+    int32_t n_mult  = 256;
+    int32_t n_head  = 32;
+    int32_t n_layer = 32;
+    int32_t n_rot   = 64;
+    int32_t f16     = 1;
+};
+
+struct llama_context;
+
+void llama_free_context(llama_context* ctx);
+
+const std::vector<gpt_vocab::id>& llama_context_get_embd(const llama_context& ctx);
```

Comment:
Please expand this unless there’s a strong reason to shorten it:

```suggestion
const std::vector<gpt_vocab::id>& llama_context_get_embedding(const llama_context& ctx);
```

Reply:
Updated.